cogboard-app/src/main/kotlin/com/cognifide/cogboard/http/HttpClient.kt

```diff
@@ -88,34 +92,63 @@ class HttpClient : AbstractVerticle() {
         val pass = config.getString(CogboardConstants.PROP_PASSWORD) ?: ""
         val token = config.getString(CogboardConstants.PROP_TOKEN) ?: ""
         val headers = config.getJsonObject(CogboardConstants.PROP_HEADERS)
+        val authenticationTypes = Json.decodeValue(config.getString(CogboardConstants.PROP_AUTHENTICATION_TYPES))
+                ?: JsonArray()
 
-        if (user.isNotBlank() && token.isNotBlank()) {
-            request.basicAuthentication(user, token)
-        } else if (user.isNotBlank() && pass.isNotBlank()) {
-            request.basicAuthentication(user, pass)
+        val authenticationType = getAuthenticationType(authenticationTypes as JsonArray, user, token, pass)
+
+        if (authenticationType == AuthenticationType.NONE) {
+            throw Exception("Request cannot be processed. Cause: Incorrect credentials or Widget doesn't supported")
+        } else {
+            AuthenticationFactory(user, pass, token, request).create(authenticationType)
```

Comment:
Instead of creating this class you could use extensiom method and call:
`request.authenticate(authenticationType, user, pass, token)`